### PR TITLE
Add the option for choosing the number of comments to show and deleti…

### PR DESCRIPTION
**Pull request by jiaxi312**
…ng comment.

- Add the option for choosing the max number of comments to show.

- Add the option for delete a selected comment.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -36,13 +37,21 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Decide the max number of comments to display
+    final int maxComments = Integer.parseInt(
+                getParameterWithDefault(request, "comments-to-show", DEFALUT_COMMENTS_TO_SHOW));
+
     // Fetch the comment from the datastore
     PreparedQuery comments = DEFAULT_DATASTORE_SERVICE.prepare(new Query("Comment"));
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: consider to use https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.Builder.html#withlimit

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -89,4 +104,20 @@ private String getParameterWithDefault(HttpServletRequest request, String name,
   private static final DatastoreService DEFAULT_DATASTORE_SERVICE 
                           = DatastoreServiceFactory.getDatastoreService();
   private static final Gson GSON = new Gson();
+  private static final String DEFALUT_COMMENTS_TO_SHOW = "10";
+
+  /** Class that stores the content of a comment and it's id in the datastore */
+  private static class Comment {
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: private static final class?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -89,4 +104,20 @@ private String getParameterWithDefault(HttpServletRequest request, String name,
...
+
+    /** Construct a comment with given content and id*/
+    private Comment(long id, String content) {
+        this.content = content;
+        this.id = id;
+    }
+
+    private final String content;
+    private final long id;
+
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: why new line here?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -89,4 +103,18 @@ private String getParameterWithDefault(HttpServletRequest request, String name,
   private static final DatastoreService DEFAULT_DATASTORE_SERVICE 
                           = DatastoreServiceFactory.getDatastoreService();
   private static final Gson GSON = new Gson();
+  private static final String DEFALUT_COMMENTS_TO_SHOW = "0";
+
+  /** Class that stores the content of a comment and it's id in the datastore */
+  private static final class Comment {
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: could you ask virtual support for the best practice on such use cases?
I doubt if it provides good readability. (In c++, we normally put such classes right before the functions that use it)

Comment (jiaxi312) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I found a better way to do that without using a nested class.
Instead, I passed a list of entities in the doGet, and change some code in script.js.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -21,9 +21,13 @@ function createListElement(text) {
 
 /** Fetches comments from the datastore and adds them to DOM */
 function loadComments() {
-    fetch('/data').then(response => response.json()).then((comments) => {
-        // Display those comments as a list
+    const numComments = document.getElementById('comments-to-show').value;
```
Comment (zixiyan) [inline portfolio/src/main/webapp/script.js]: per https://google.github.io/styleguide/jsguide.html#formatting-block-indentation, use 2 spaces indentation?

**Review by zixiyan — APPROVED** [review]

**Pull request merged**